SetupDataPkg/Library/ConfigKnobShimLib/ConfigKnobShimPeiLib/GoogleTest/ConfigKnobShimPeiLibGoogleTest.cpp

```diff
@@ -0,0 +1,217 @@
+/** @file ConfigKnobShimPeiLibGoogleTest.cpp
+  This is a sample to demonstrates the use of GoogleTest that supports host
+  execution environments.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Library/GoogleTestLib.h>
+#include <GoogleTest/Library/MockUefiRuntimeServicesTableLib.h>
+#include <GoogleTest/Library/MockPeiServicesLib.h>
+#include <GoogleTest/Ppi/MockReadOnlyVariable2.h>
+
+extern "C" {
+#include <Uefi.h>
+#include <Library/BaseLib.h>
+#include <Library/DebugLib.h>
+
+#include "../../ConfigKnobShimLibCommon.c"   // include the c file to be able to unit test static function

```

Comment:
could the source file be included in the INF file's section?
Then just add the C file declarations here?

Reply:
Note for the PR: Changed this so the source file is included in the INF file's section, and the ConfigKnobShimLib.h is included here in the test file.

Reply:
That won't work, because a static function is being tested. So linking won't work here, we actually need to include the c file, which is hacky, but it lets us test static functions.

Reply:
Including in the inf sources did work for me in this case.